src/main/java/org/broadinstitute/hellbender/utils/variant/VariantContextVariantAdapter.java

```diff
@@ -24,9 +35,48 @@
     public boolean isSnp() { return variantContext.isSNP(); }
     @Override
     public boolean isIndel() { return variantContext.isIndel(); }
+
+    @Override
+    public UUID getUUID() {
+        return uuid;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
```

Comment:
What we really want in implementations of `Variant`, I think, is what we already have for `GATKRead`: both a strict `equals()` that checks everything including UUID, as well as an `equalsIgnoreUUID()` that checks for value equality and ignores the UUIDs. `VariantUtils.variantsAreEquals` could be the value equality checker that both of these would call into.